### PR TITLE
Adds more logger tests. Tweaks some options with the logger.

### PR DESCRIPTION
**Pull request by wcharczuk**
## PR Summary

*Replace this text with a description of what this PR is accomplishing*

 - **Type:** Internal
 - **Issue Link:** N/A
 - **Intended Change Level:** patch

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

File: logger/COVERAGE [hunk 1]
```diff
@@ -1 +1 @@
-46.9
\ No newline at end of file
+39.81
```
Comment (wcharczuk) [inline logger/COVERAGE]: @gcarling wat

Comment (mikes-blend) [reply logger/COVERAGE]: 🤔

File: logger/default.go [hunk 2]
```diff
@@ -6,48 +6,48 @@ import (
 )
 
 var (
-	log     *Logger
-	logInit sync.Once
+	_log     *Logger
```
Comment (mikes-blend) [inline logger/default.go]: What is the convention for prepending underscores to var names?

**Comment by wcharczuk** [comment]
> When did the logger singleton come back? I thought we were trying to move away form the singleton pattern within the SDK libs?

it got added back as a compatibility shim for paul (it mirrors the api for sirupsen/logrus). we can remove

File: logger/default.go (hunk 2 above)
Comment (wcharczuk) [reply logger/default.go]: in this case, it avoids a conflict with the stdlib `log` package. the linter (golint) allows it, but we probably shouldn't promulgate it as a pattern.

**Pull request merged**